scrape/images.ts

```diff
@@ -0,0 +1,64 @@
+import { DataImage } from "../types/types"
+import { manuallyListedImages } from "../data/images-manual"
+import { scrapedImages } from "../data/images-scraped"
+import { } from "bun"
+import { join } from "node:path"
+
+export const getImages = async (): Promise<DataImage[]> => [
+    ...manuallyListedImages,
+
+    ...await (async () => {
+        const icons: DataImage[] = []
+
+        await Promise.all(scrapedImages.map(async (scrapedImage) => {
+            try {
+                await Bun.$`git clone https://github.com/${scrapedImage.ownerRepoPath}.git ./cloned/${scrapedImage.ownerRepoPath}`.cwd("./scrape")
+            } catch {
+                // pull instead
+                await Bun.$`git pull`.cwd(`scrape/cloned/${scrapedImage.ownerRepoPath}`)
+            }
+
+            const branch = (await Bun.$`cd ./ && git branch --show-current`.cwd(`scrape/cloned/${scrapedImage.ownerRepoPath}`).text()).replace("\n", "")
+
+            const glob = new Bun.Glob('**/*.{png,svg,jpg,jpeg,gif,webp,avif,apng,tiff}')
+            for await (const file of glob.scan({ cwd: join(import.meta.dir, "./cloned/", scrapedImage.ownerRepoPath) })) {
+                icons.push({
+                    author: scrapedImage.author,
+                    className: scrapedImage.className,
+                    title: file.split("/").pop()!,
+                    src: `https://raw.githubusercontent.com/${scrapedImage.ownerRepoPath}/${branch}/${file}`,
+                    raw: `https://github.com/${scrapedImage.ownerRepoPath}/blob/${branch}/${file}`,
+                })
+            }
+        }))
+
+        return icons.sort((a, b) => a.title.localeCompare(b.title))
```

Comment:
i sorted it by something just so the diff wont be broken (with everything changing order), but i think in future we can get the initial commit time and sort by that.